frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -158,37 +169,55 @@ export const insightLogic = kea<insightLogicType>({
                 setTimeout(() => {
                     if (values && view == values.activeView) {
                         actions.setShowTimeoutMessage(true)
+                        const tags = {
+                            insight: values.activeView,
+                            scene: sceneLogic.values.scene,
+                        }
+                        posthog.capture('insight timeout message shown', tags)
+                        captureInternalMetric({ method: 'incr', metric: 'insight_timeout', value: 1, tags })
                     }
                 }, SHOW_TIMEOUT_MESSAGE_AFTER)
             )
             actions.setIsLoading(true)
         },
-        endQuery: ({ view, lastRefresh, exception }) => {
-            clearTimeout(values.timeout || undefined)
+        endQuery: ({ queryId, view, lastRefresh, exception }) => {
+            if (values.timeout) {
+                clearTimeout(values.timeout)
+            }
             if (view === values.activeView) {
                 actions.setShowTimeoutMessage(values.maybeShowTimeoutMessage)
                 actions.setShowErrorMessage(values.maybeShowErrorMessage)
                 actions.setLastRefresh(lastRefresh || null)
                 actions.setIsLoading(false)
-                if (values.maybeShowTimeoutMessage) {
```

Comment:
look like we're not using this anywhere anymore, let's remove it from above and in reducers

Reply:
It seems to be used above (see line `actions.setShowTimeoutMessage(values.maybeShowTimeoutMessage)`)

Reply:
oh, I misread how things work, thanks